### PR TITLE
Features/task 850 create a concept chapter for state machines

### LINKED ISSUE #850: Create a Concept Chapter for State Machines in the User Manual
**Description**
A clear and concise description of what this ticket is about.

### PR DESCRIPTION
**Pull request by GeorgeNasrDLR**
Just Nothing

Closes #850

### CODE REVIEW

**Review by GeorgeNasrDLR — COMMENTED** [review]
done

**Review by GeorgeNasrDLR — COMMENTED** [review]
done

**Review by Anna-Lena-DLR — APPROVED** [review]

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc [hunk 1]
```diff
@@ -1232,6 +1232,112 @@ image:chapterModeling/MaturityOverview.png[Maturity overview]
...
+
+The basic building blocks of a state machine are:
+
+* `State:` 
+is a situation of a system depending on previous inputs and causes a reaction on following inputs. One state is marked as the initial state; this is where the execution of the machine starts.
+* `Transition:` defines for which input a state is changed from one to another. Depending on the state machine type, states and/or transitions produce outputs.
+* `AllowsConstraint:` should specify the states that are allowed to be activated while the parent state is active.
+* `ForbidsConstraint:` should specify the states that are forbidden while the parent state is active.
+
+In the following we will create a mass budget. Create a project - in the following called `MySatellite` - and activate the following concepts
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc:1248]: "we will create a mass budget" -> "we will create a state machine" ?

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc [hunk 2]
```diff
@@ -1232,6 +1232,112 @@ image:chapterModeling/MaturityOverview.png[Maturity overview]
...
+|Open the _Diagram Editor_ and add the following _States_ and _Transitions_ as shown in the image. This shall serve as an example to represent the behavior of the `AOCS` system.
+
+NOTE: The green and red arrows shown in the state machine shall represent the _AllowsConstraints_ and _ForbidsConstraints_ respectively.  For instance, the thrusters are on when the safe mode of the satellite is active to perform coarse pointing control.
+
+TIP: To use the _State Machine_ of other system components, e.g. the Camera, Reaction Wheels in Main system _State Machine_, one can just drag and drop them from the _Navigator_. 
+
+|image:chapterStateMachines/AOCSSystemStateMachine.png[AOCSSystemStateMachine] image:chapterStateMachines/AOCSTransitionsAndTriggerEvents.png[AOCSTransitionsAndTriggerEvents]
+
+|{counter:step1}.
+|For each section in _State Machine_ concept found in the Editor an excel table with the section instances can be created. This shown in the following image.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc:1337]: "This shown in the following image" -> This is shown in the following image"

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc [hunk 3]
```diff
@@ -1232,6 +1232,112 @@ image:chapterModeling/MaturityOverview.png[Maturity overview]
...
+|Now having created the various components of the system and described their behavior using state machines, the _State Machine_ for the `AOCS` system can be created in a newly created _ElementConfiguration_ namely `CONOPS` (standing for concept operations). This state machine shall describe the behavior of the whole system.
+
+See Task No.4 “Creating a State Machine”
+
+|image:chapterStateMachines/AddingSystemStateMachine.png[AddingSystemStateMachine]
+
+|{counter:step1}.
+|Open the _Diagram Editor_ and add the following _States_ and _Transitions_ as shown in the image. This shall serve as an example to represent the behavior of the `AOCS` system.
+
+NOTE: The green and red arrows shown in the state machine shall represent the _AllowsConstraints_ and _ForbidsConstraints_ respectively.  For instance, the thrusters are on when the safe mode of the satellite is active to perform coarse pointing control.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc:1330]: We should try to do 1 sentence per line. Asciidoc will handle the layouting.

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc [hunk 4]
```diff
@@ -1232,6 +1232,112 @@ image:chapterModeling/MaturityOverview.png[Maturity overview]
...
+NOTE: Transitions can be created or removed in the _Diagram Editor_, but also manually in the normal _Editor_.
+
+|image:chapterStateMachines/AssigningTriggers.png[AssigningTriggers] image:chapterStateMachines/OpenedWizardForTriggerActions.png[OpenedWizardForTriggerActions]
+
+|{counter:step1}.
+|After assigning, go back to the _Diagram Editor_ and update the state machine diagram using the yellow button up right the _State Machine_ block, then you will be able to see the new transition.
+|image:chapterStateMachines/UpdateStateMachine.png[UpdateStateMachine]
+
+|{counter:step1}.
+|Now having created the various components of the system and described their behavior using state machines, the _State Machine_ for the `AOCS` system can be created in a newly created _ElementConfiguration_ namely `CONOPS` (standing for concept operations). This state machine shall describe the behavior of the whole system.
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc:1321]: We shouldnt use "CONOPS" in the example. Try to find a different name here. The reason is that CONOPS is a critical term for export control and might cause issues.

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
Thats really great work! Just minor things.

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc (hunk 2 above)
Comment (GeorgeNasrDLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc]: changed

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc (hunk 3 above)
Comment (GeorgeNasrDLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc]: changed

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc (hunk 4 above)
Comment (GeorgeNasrDLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc]: changed

File: de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc (hunk 1 above)
Comment (GeorgeNasrDLR) [reply de.dlr.sc.virsat.docs.feature/src/docs/VirSat_Core_User_Manual.adoc]: changed

**Comment by SaMuellerDLR** [comment]
@GeorgeNasrDLR Please also update the description of the pull request. It still says WIP.

**Comment by franzTobiasDLR** [comment]
Your screenshots have the interlacing flag on... That's not support by our asciiDoc generator. That's why the build is failing. Open them with gimp and directly export them with the "interlacing" flag unchecked again to fix the problem

**Comment by GeorgeNasrDLR** [comment]
Yes Tobi. I actually misunderstood Paulina considering this, and unchecked the Interlacing thing when it was checked and vice versa. So thanks for your comment. It helped

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Pull request merged**